ledger/simulation/simulator.go

```diff
@@ -243,6 +244,18 @@ func (s Simulator) Simulate(simulateRequest Request) (Result, error) {
 		}
 	}
 
+	if simulatorTracer.result.EvalOverrides.AllowUnnamedResources {
+		// Remove private fields for easier test comparison
+		simulatorTracer.result.TxnGroups[0].UnnamedResources.removePrivateFields()
+		for i := range simulatorTracer.result.TxnGroups[0].Txns {
+			txnResult := &simulatorTracer.result.TxnGroups[0].Txns[i]
+			if !txnResult.UnnamedResources.HasResources() {
+				// Clean up any unused local resource assignments
+				txnResult.UnnamedResources = nil
```

Comment:
Yes -- that loop you referenced assigns a value to `groupResult.Txns[i].UnnamedResources` for each transaction. However if the local unnamed resource assignment for a transaction is never used (e.g. because it's not an app call, or it's an app call that uses group sharing instead), I want to remove it from the response, which is why we have the `HasResources()` check.

I thought it would be simpler for the responses if unused resource assignments were simply omitted. Right now this can only happen for the local txn level, but maybe we could check and remove group-level objects too?

Reply:
traversing back and forth on endpoint response level code, and I think it might be a good time to answer this month old question.

I would up the vote for remove group level objects, if I understood correctly, which is what you are doing in `v2/utils.go`, that applies `convertUnnamedResourceAssignment` to the aggregration all group elements (app-local, account-ref, box).

If you think my observation above is making sense, then we are good to resolve this thread.